src/core/lib/channel/channel_stack.cc

```diff
@@ -117,8 +119,12 @@ grpc_error_handle grpc_channel_stack_init(
     }
   }
 
+  auto conditioned_channel_args = grpc_core::CoreConfiguration::Get()
```

Comment:
In principal, I think this preconditioning should be done where the channel args are created, not where they are used.  How many places do we create new channel args within C-core?

We should also consider why we are creating channel args within C-core in the first place.  In principal, there should be very few places where this happens -- it should generally be necessary only when constructing a control-plane channel within C-core, such as the xDS channel, the grpclb channel, or the ALTS channel, and those places should get preconditioning automatically via `grpc_channel_create()`.  If there are other places where we're doing this, I think we should look at each one and decide whether it's actually necessary.

For example, do we actually have any cases where we use `ChannelStackBuilder` without calling `SetChannelArgs()`?  If not, maybe we should just change the `ChannelStackBuilder` API to require passing in channel args to its ctor, so that this case goes away.

I don't think we want the preconditioning to happen whenever we instantiate a `ChannelArgs` object, because those objects get copied all the time within C-core.

Reply:
> How many places do we create new channel args within C-core?

To begin with, I count around 30 default-constructed ChannelArgs class members within src/core. I haven't gone through to see how many get overwritten, or could otherwise be accessed before being overwritten - LB policies and client channel resolvers make up about a third of the set. I'll take a deeper look here.

I did the experiment you suggested with ChannelStackBuilder, indeed we can add a ChannelArgs argument to the constructor, and all PR tests pass. So we may be safe with that change alone.

> I don't think we want the preconditioning to happen whenever we instantiate a ChannelArgs object, because those objects get copied all the time within C-core.

... and we'd be adding a dependency cycle between core config and channel args, as well as making making channel args fragments more expensive (as used in resolvers). It was a thought, but wouldn't pan out right now.